app/tests/spec/lib/channel-server-client.js

```diff
@@ -5,55 +5,93 @@
 import a256gcm from 'lib/crypto/a256gcm';
 import { assert } from 'chai';
 import ChannelServerClient from 'lib/channel-server-client';
+import { DEVICE_PAIRING_CHANNEL_KEY_BYTES } from 'lib/constants';
 import sinon from 'sinon';
 import { wrapAssertion } from '../../lib/helpers';
 import ChannelServerClientErrors from '../../../scripts/lib/channel-server-client-errors';
 
 describe('lib/channel-server-client', () => {
   let client;
-  let notifier;
   let sandbox;
 
   beforeEach(() => {
-    notifier = {
-      trigger: sinon.spy()
-    };
-
     sandbox = sinon.sandbox.create();
 
     client = new ChannelServerClient({
       channelId: 'channel-id',
       channelKey: 'channel-key',
       channelServerUrl: 'wss://channel.server.url/',
-    }, {
-      notifier,
     });
   });
 
   afterEach(() => {
     sandbox.restore();
   });
 
+  describe('_deriveEncryptionKey', () => {
+    it('resolves to the encryption key', () => {
+      const channelKeyBuffer = Buffer.from('channel key', 'utf8');
+      const channelIdBuffer = Buffer.from('channel id', 'utf8');
+      return client._deriveEncryptionKey(channelKeyBuffer, channelIdBuffer)
+        .then(keyBuffer => {
+          assert.lengthOf(keyBuffer, DEVICE_PAIRING_CHANNEL_KEY_BYTES);
+          // python snippet to generate result:
+          // >>> from tokenlib.utils import HKDF
+          // >>> HKDF("channel key", "channel id", "identity.mozilla.com/picl/v1/pair/encryption-key", 32).encode('hex')
+          // '0d91a09d8e04f34cdd3045026eb98d96c3e999908f58956fdba6a39b685472f8'
+          assert.equal(keyBuffer.toString('hex'), '0d91a09d8e04f34cdd3045026eb98d96c3e999908f58956fdba6a39b685472f8');
+        });
+    });
+  });
+
   describe('_deriveChannelJwk', () => {
     it('creates a Jwk', () => {
-      const channelKeyBuffer = Buffer.from('channel id', 'utf8');
-      const channelIdBuffer = Buffer.from('channel key', 'utf8');
-      return client._deriveChannelJwk(channelKeyBuffer, channelIdBuffer)
-        .then(jwk => {
-          assert.equal(jwk.alg, 'A256GCM');
-          assert.equal(jwk.kty, 'oct');
-          assert.ok(jwk.kid);
+      const channelKeyBuffer = Buffer.from('channel key', 'utf8');
+      const channelIdBuffer = Buffer.from('channel id', 'utf8');
+      return Promise.all([
+        client._deriveChannelJwk(channelKeyBuffer, channelIdBuffer),
+        client._deriveEncryptionKey(channelKeyBuffer, channelIdBuffer),
+      ]).then(([jwk, expectedKeyBuffer
+      ]) => {
+        assert.equal(jwk.length, 256);
+
+        const jwkObj = jwk.toObject(true);
+        assert.equal(jwkObj.alg, 'A256GCM');
+        assert.equal(jwkObj.kty, 'oct');
+        assert.equal(jwkObj.k.toString('hex'), expectedKeyBuffer.toString('hex'));
+        assert.ok(jwk.kid);
+      });
+    });
+  });
+
+  describe('_deriveEncryptionKey', () => {
```

Comment:
This appears to be identical to the `_deriveEncryptionKey' test above, is it an accidental dup?

Reply:
It is.